Coding convention applied - part 1

Quix0r opened the pull request:
I have again applied more coding convention stuff, e.g. added missing curly braces, spaces and usage of `dbm::is_result()`. Also I have unified most db upgrades to a more common way:

````
function update_xxxx() {
    $r = q("ALTER TABLE `foo` DO SOMETHING");

    if ($r) {
        return UPDATE_SUCCESS;
    }
    return UPDATE_FAILED;
}
````
There are a few exceptions, plus a lot older updates that needs being fixed/considered.

Rebased #3103 + this later:

More similar changes: (in order they are created)

- [ ] part 2 https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-split2
- [ ] part 3 https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-split3
- [ ] part 4 https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-split4
- [ ] part 5 https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-split5
- [ ] part 6 https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-split6
- [ ] / : https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-root
- [ ] utils/ : https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-utils
- [ ] mod/ https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-mod
- [ ] include 1: https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-include-if
- [ ] include 2: https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-include-foreach
- [ ] others: https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-others
- [ ] else-if: https://github.com/Quix0r/friendica/tree/rewrites/coding-convention-else-if

Unrelated but comes next:
- [ ] chmod https://github.com/Quix0r/friendica/tree/fixes/chmod

Hypolite commented:
@Quix0r How do you use the checkboxes in the PR original comment? It shows as a task list in other displays:
![screen shot 2017-01-27 at 11 56 11 am](https://cloud.githubusercontent.com/assets/925415/22379513/a47e3e2c-e487-11e6-992e-395da44c8b3f.png)

rabuzarus commented:
https://github.com/blog/1825-task-lists-in-all-markdown-documents

annando commented:
Why have you remade this pull request?

tobiasd commented:
@annando for `git magical reasons` see the last comment in #3103 

> Argh, rebasing + push delete seems to close this. :-(

Quix0r commented:
Yeah, sorry for the mess. :-( I used these checkboxes to keep track in which order I have to make new PRs once the current is done. @Hypolite @annando yes, the PR was "auto-magically" closed by github because the underlaying branch was gone. Now I cannot reopen it.

Quix0r commented:
I am rebasing because I don't want so much merges in git history. A lot merges make the git history complex (according to documentation). So if there are conflicting commits, a good choice is to rebase over merging.

Quix0r commented:
Rebased and force-pushed my PR (`git push --all --force`).

annando commented on include/dfrn.php:
```diff
@@ -817,46 +877,61 @@ private function entry($doc, $type, $item, $owner, $comment = false, $cid = 0) {
 				dbesc(normalise_link($mention)));
-			if ($r[0]["forum"] OR $r[0]["prv"])
+
+			if (!dbm::is_result($r)) {
```
I would do this differently. Quitting here is something I wouldn't prefer.

Quix0r replied:
`dbm::is_result()` is for determining if a result has been returned, not if rows has (>=1 count) has been returned. Maybe I need to introduce another method, e.g. `dbm::has_rows()` (I prefer camel-case for method names here but I go with this project's style rule).

Quix0r replied:
I like to differenciate between failed SQL request (e.g. bad syntax, gone database connection) and no rows (but valid SQL request) returned.

annando commented on include/dfrn.php:
```diff
@@ -873,23 +948,28 @@ private function entry($doc, $type, $item, $owner, $comment = false, $cid = 0) {
+
 		// use RINO1 if mcrypt isn't installed and RINO2 was selected
-		if ($rino==2 and !function_exists('mcrypt_create_iv')) $rino=1;
+		if ($rino==2 and !function_exists('mcrypt_create_iv')) {
```
Spaces before and behind ````==```` would be better.

Quix0r replied:
In this PR I was not looking for such things to make lesser changes to an already large change. I would prefer to change such things "batched" later on and mix with this (adding curly braces mostly).

annando replied:
I would prefer to have a completely cleaned file in one task. But I don't see this as a blocker.

Quix0r replied:
Okay, was not much change.

annando commented on include/dfrn.php:
```diff
@@ -1475,8 +1602,10 @@ private function process_suggestion($xpath, $suggestion, $importer) {
 			intval($suggest["uid"])
 		);
-		if (dbm::is_result($r))
+		/// @TODO Really abort on valid result??? Maybe missed ! here?
```
Yeah, it is intended. See the comment above.

annando commented on include/dfrn.php:
```diff
@@ -1494,10 +1623,12 @@ private function process_suggestion($xpath, $suggestion, $importer) {
 				intval($fid)
 			);
-			if (dbm::is_result($r))
+			/// @TODO Really abort on valid result??? Maybe missed ! here?
```
Yeah, it is intended. See the comment above.

Quix0r replied:
Removed my TODO and added regular comment.

annando commented on include/dfrn.php:
```diff
@@ -1555,11 +1687,13 @@ private function process_suggestion($xpath, $suggestion, $importer) {
 
 		logger("Processing relocations");
 
+		/// @TODO Rewrite this to one statement
```
Why?

Quix0r replied:
It makes code be easier and is a common rule:
````
if ($foo == 'for') {
if ($bar != 'bar') {
doFooNoBar();
}
}
````
... can be rewritten to:
````
if ($foo == 'for' && $bar != 'bar') {
doFooNoBar();
}
````
... is much smaller and lesser nested curly braces.

Quix0r replied:
Ah, to lazy to check code. I like the one-array approach:
````
$someArray = array(
    'bar'      => 'bar_value',
    'foo'      => 'foo_value',
    'bar_baz' => 'baz_foo',
);
````

annando commented on include/ostatus.php:
```diff
@@ -37,59 +37,70 @@ class ostatus {
 	 */
 	private function fetchauthor($xpath, $context, $importer, &$contact, $onlyfetch) {
 
+		/// @TODO One statment is enough ...
```
I must admit: I prefer that style.

Quix0r replied:
See my example above.

annando commented on include/ostatus.php:
```diff
@@ -759,11 +824,12 @@ private function get_actor_details($actor, $uid, $contact_id) {
+		}
 
-		if ($contact) {
+		if (dbm::is_resul($contact)) {
```
You forgot a "t".

Quix0r replied:
Fixed.

annando commented on include/ostatus.php:
```diff
@@ -897,13 +970,19 @@ private function completion($conversation_url, $uid, $item = array(), $self = ""
 
 		$r = q("SELECT `nurl` FROM `contact` WHERE `uid` = %d AND `self`", intval($uid));
+
+		if (!dbm::is_result($r)) {
```
Quitting silently is something that I don't prefer.

Quix0r replied:
I added a TODO as I don't know what I should take here.

annando replied:
A to-do doesn't help anyone here. I would rather prefer to comment out these added checks and to add a to-do as a reminder for the future to verify if they could be harmful under certain circumstances.

Quix0r replied:
Added some logging here, too.

annando commented on include/ostatus.php:
```diff
@@ -1050,6 +1138,7 @@ private function completion($conversation_url, $uid, $item = array(), $self = ""
 				continue;
 			}
 
+			/// @TODO One statment is okay (until if() )
```
I really, really prefer it that way.

Quix0r replied:
No, it is commonly not a good practice, to have tons of nested curly braces around as it makes the code more complicated (to read/review by human not to parse by program).

annando commented on mod/settings.php:
```diff
@@ -364,41 +364,44 @@ function settings_post(App $a) {
+		//  changing it to the new value
+		$r = q("SELECT `password` FROM `user`WHERE `uid` = %d LIMIT 1", intval(local_user()));
+		if (!dbm::is_result($r)) {
+			killme();
```
When we quit here, we need to log it.

Quix0r replied:
Will be included in next PR. I have had already added a `@TODO` so it won't be forgotten.

annando replied:
I still have a bad feeling about this. Isn't it possible to add a ```logger("User ".local_user()." wasn't found, quitting here", LOGGER_DEBUG);``` with this PR?

Quix0r replied:
Okay, if requested, I go ahead and add it. :-)

annando commented on update.php:
```diff
@@ -1210,114 +1223,153 @@ function update_1136() {
+			`id` INT NOT NULL AUTO_INCREMENT PRIMARY KEY ,
+			`name` CHAR( 128 ) NOT NULL ,
+			`locked` TINYINT( 1 ) NOT NULL DEFAULT '0'
 		  ) ENGINE = MYISAM DEFAULT CHARSET=utf8 ");
```
We should remove the "ENGINE = MYISAM" from the update since we now prefer using InnoDB.

Quix0r replied:
There are 4 identical statements in the same file. Should I go and remove them? (separate commit, of course)

annando replied:
Please remove them in any possible way :)

Quix0r replied:
Should be gone now.

annando commented on update.php:
```diff
@@ -1451,14 +1523,17 @@ function update_1163() {
 function update_1164() {
 	set_config('system', 'maintenance', 1);
 
+	/// @TODO If one update fails, should it continue?
```
Why not?

Quix0r replied:
Maybe other updates depend on a single update?

Quix0r replied:
I think, I put further rewrites towards unified database updates in an other branch/PR and not mix with this (coding-convention and unifying updates are two different things).

Quix0r replied:
Now these changes are in an other branch, thanks to resetting hard to a specific commit:

````
$ git checkout rewrites/coding-convention
$ git checkout -b rewrites/coding-convention-splitX
$ git checkout rewrites/coding-convention
$ git reset --hard xyz123
````
This is what you can do to go back in a branch and still preserve the other changes.

annando replied:
I wouldn't invest much time into this update functionality. Since a long time we are using another functionality. And even if the database update scripts here do fail, there is always the dbstructure script.

Quix0r replied:
Removed my TODO.

Quix0r commented:
Further rebase done, was not easy. :-)

Quix0r commented:
Damn, `--force` from workspace has killed them. :-( I need to push from home again ...

annando commented:
I guess you really should avoid doing a rebase or something similar. I don't know what you did, but now this pull request contains **193** changed files - which is really monsterous.

I see a huge problem with this pull request. Sadly it does not only contain changes to the coding conventions alone. It does also contain some changes to the inner logic. (See the added checks for empty database results)

Going through 193 files to check if the changes had done no harm is a very long process that will surely last for some hours. But the additional checks if the newly inserted code could cause some harm could bind me for some more hours.

And then is the problem that with every requested change I have to go through the whole changes again.

I really would prefer to split this monster into several pull requests with no more than 20 files. And I really, really would prefer to separate the style changes from the logic changes.

Can this be done with some git magic?

I see the huge problem that this pull request is mostly blocking the release of the 3.5.1. All pull requests for the 3.5.1 are done and (nearly) all issues have been solved.

I have the feeling that the changes in this pull request are too huge to be sure that they don't introduce new problems. So I would suggest not only the split in maybe 10 single pull request but I would also suggest to postpone it to the 3.5.2.

What are @Hypolite, @rabuzarus, @fabrixxm and @tobiasd think?

rabuzarus commented:
I'm unsure. Would it be OK for you @Quix0r ?

At the moment I don't have the time to review such big PR's. 

One note about such PR's. This does behave a little bit like a circle. @Quix0r does a really good job by applying the coding conventions or by replacing outdated functions/code snippets. I'm really happy that someone is doing the work which is often annoying. We have many of such parts so the PR is often bigger than 2 or 3 files. The bigger the PR the more we need to review. And so it lasts until it gets accepted. In the meanwhile other PR's get accepted and @Quix0r needs to rebase the code (or merge develop) and maybe he needs to adjust some changed parts again. I guess this is also frustrating. After this we need to review the whole PR again to make sure there is no mistake in this work. This needs time and is frustration because we have read such a big PR some time ago. And it takes time.
So the circle starts from the beginning.

From my point of view there are two possibilities.
1.) Smaller PR's or
2.) We give such PR's a priority. This would mean we try to review it very quickly (3 or 4 days) and don't accept other PR's which would touch the changed files of the big PR

Quix0r commented:
Fine with me to rebase my work and force-push it (`git push --force`). I can be very patient. :-) Yeah, I know these PRs are important but can become very large. I already splitted these changes into smaller "chunks" - see my linked branches above, I need to rebase them before I submit a PR.

So should I split them more, let's say max. 10 files per PR?

Quix0r commented:
So now I know it better, don't delete the branch, use `--force` while pushing conflicting (rebased) changes.

annando commented:
Splitting it does help a lot since this will increase the possibility that more of the PR are accepted very fast. There will - however - be some PRs that will still be open for some more days. But they then can iterate without too much collateral damage.

Quix0r commented:
F*ck, why so much new commits has been added? :-(

Quix0r commented:
Rebasing seems to re-add the commits also here in web frontend. I hope you don't have to review all over again ... :-( *sigh*

annando commented:
I have the feeling that "rebase" is evil and should be avoided at any time ...

Hypolite commented:
Rebase isn't evil but it is pretty inconvenient as soon as you're trying to share your changes, and I would agree that it should be avoided.

Quix0r commented:
It looks like that github is depending on commit ids to show what needs to be reviewed (which is convincing). `rebase [-S]` re-creates those commit ids as the changes are being applied again and therefor a new hash is being produced (new parent hash and/or entropy).

So how do we want to handle these kind of commits? Yes, in one side they need to be done (as many is not consistent in Friendica) but on other hand rebasing often is (usually) common but here causing reviewers to start over.

Quix0r commented:
One idea:

1. postpone (not reject) PR (PRs must be done by separate branches)
2. at the right time, ask to rebase the branch to have it "in sync" with latest changes
3. Do the rebase. This also gives the developer a change to self-review his commits
4. ask to review now (not before) as the commit ids have changed

annando commented:
Whatever is done - we have to make sure that on github there will only be a handful of files to review per PR. 193 files are much too high :)

Quix0r commented:
I can re-create my PR, maybe, and split it up into more commits, if possible.

Quix0r commented:
I have now reset to https://github.com/friendica/friendica/pull/3112/commits/3d591fd9a7a0c697202fc9851946bce167b3f454 but still the last commit here is https://github.com/friendica/friendica/pull/3112/commits/59acf2f3c153cba374ab9d8e644fe353f0ecab6e

```
$ git status
On branch rewrites/coding-convention
Your branch is up-to-date with 'origin/rewrites/coding-convention'.
nothing to commit, working directory clean
$
```
and ...
```
$ git show
commit 3d591fd9a7a0c697202fc9851946bce167b3f454
Author: Roland Häder <roland@mxchange.org>
Date:   Wed Jan 25 16:02:02 2017 +0100

    Opps ...

    Signed-off-by: Roland Häder <roland@mxchange.org>

diff --git a/include/dfrn.php b/include/dfrn.php
index 15e2fc3..9ea0daa 100644
--- a/include/dfrn.php
+++ b/include/dfrn.php
@@ -1131,7 +1131,7 @@ class dfrn {


                        if ($dfrn_version >= 2.1) {
-                               if (($contact['duplex'] && strlen($contact['pubkey'])) {
+                               if (($contact['duplex'] && strlen($contact['pubkey']))
                                        || ($owner['page-flags'] == PAGE_COMMUNITY && strlen($contact['pubkey']))
                                        || ($contact['rel'] == CONTACT_IS_SHARING && strlen($contact['pubkey']))) {


$
```

annando commented:
Whatever you are doing: Please have a look at the number of "files changed" at the top of this page https://github.com/friendica/friendica/pull/3112

As long as there are still 193 files, there is a problem.

Quix0r commented:
Here is what I have:
````
$ git branch -v
* develop                                    10bfc46 added spaces + some curly braces + some usage of dbm::is_result()
  fixes/chmod                                943e72f Merge remote-tracking branch 'upstream/develop' into develop
  rewrites/coding-convention                 3d591fd Opps ...
  rewrites/coding-convention-else-if         6294ece Continued a bit: - converted some else if to elseif (only PHP) - converted some space -> tab (intending)
  rewrites/coding-convention-include-foreach 99c8869 more spaces + some curly spaces added
  rewrites/coding-convention-include-if      c03f4d7 added spaces + some curly braces
  rewrites/coding-convention-mod             666d86c added spaces + some curly braces + some usage of dbm::is_result()
  rewrites/coding-convention-others          d8e0384 added spaces (coding convention)
  rewrites/coding-convention-root            d29c107 added missing spaces/curly braces
  rewrites/coding-convention-split2          81c0cd7 added more spaces/curly braces
  rewrites/coding-convention-split3          d4135ac Continued a bit: - converted some else if to elseif (only PHP) - converted some space -> tab (intending)
  rewrites/coding-convention-utils           6b09286 curly braces + spaces added
$
````
So `3d591fd` is the last commit in this branch. But according to *this* website, `59acf2f` is the last one. It looks like github is buggy. Because the last commit is really `3d591fd`.

Okay, I go more back, maybe then github is lesser confused.

Quix0r commented:
Okay, that worked. Now it is down at 177 files. I go more back ...

Quix0r commented:
Down at 11 files ... :-)

annando commented:
You are my hero :-)

Hypolite commented:
How did you do that? It looks like magic to me!

Quix0r commented:
@Hypolite see here, please: https://github.com/friendica/friendica/pull/3112#discussion_r103881645

Quix0r commented:
Have rebased my work on latest develop changes.

Quix0r commented on util/po2php.php:
```diff
@@ -12,7 +12,7 @@ function po2php_run(&$argv, &$argc) {
 	$outfile = dirname($pofile)."/strings.php";
 
-	if(strstr($outfile,'util'))
+	if (strstr($outfile,'util'))
```
Here I need to add curly braces.

Quix0r commented on util/po2php.php:
```diff
@@ -57,44 +58,65 @@ function escape_s($match){
-			if ($inv) { $inv = False; $out .= '"'.$v.'"'; }
+		if ($k!="" && substr($l,0,7)=="msgstr ") {
+			if ($ink) {
+				/// @TODO false or FALSE, but not False ! And same with TRUE/true choice
```
Yeah, I fixed some of these `False` to `false`.

Hypolite replied:
You can remove the @todo line if you fixed it.

annando commented on include/dfrn.php:
```diff
@@ -1005,11 +1044,12 @@ public static function deliver($owner,$contact,$atom, $dissolve = false) {
 
 
-		if($rino>0 && $rino_remote_version>0 && (! $dissolve)) {
+		if ($rino>0 && $rino_remote_version>0 && (! $dissolve)) {
```
Spaces before and behind the ```>``` would be nice. But like I told above, this is no blocker.

Quix0r replied:
added

annando commented on mod/settings.php:
```diff
@@ -381,24 +381,27 @@ function settings_post(App $a) {
+			$err = true;
+		}
 
 		if(! $err) {
```
Here is a space missing :)

Quix0r commented:
@Hypolite yes, that will come with per-path PRs (or more splits, if to big).

annando commented:
There is an upcoming merge conflict with my pull request https://github.com/friendica/friendica/pull/3243 in the file ```dfrn.php```: https://github.com/friendica/friendica/pull/3243/files#diff-08a925e9dc328fe7ae4f3f27479a6678L1097

In which order should we accept the pull request to not cause too much merge conflicts?

Quix0r commented on include/dfrn.php:
```diff
@@ -207,6 +216,11 @@ public static function feed($dfrn_id, $owner_nick, $last_update, $direction = 0,
 			dbesc($sort)
 		);
 
+		if (!dbm::is_result($r)) {
```
The query may fail (e.g. database crashed/restarted), so better is to validate if all queries really run successful.

Quix0r commented:
fixed conflicts with upstream develop branch

annando commented:
@Hypolite could you check if you can approve this PR as well? I really want to merge it as soon as possible now. @Quix0r has always to solve the conflicts that now occur due to our overlapping programming.

Quix0r commented:
@annando Should I solve again? How about rebasing? cc @Hypolite thank you for reviewing. :-) When these PRs are through and we meet in person (e.g. FROSS?) I buy you both "a beer" (maybe more than one ;-) or not beer at all, like coffee or Club-Mate ... )

Quix0r commented:
Oh, do you use `NULL_DATE` or `'0000-00-00 00:00:00'` ?

Quix0r commented:
Answering own question:
````
$ git blame boot.php |grep NULL_DATE
Blaming lines: 100% (2638/2638), done.
222550e7 (Michael                    2017-03-18 11:42:54 +0000  448) define ('NULL_DATE', '0001-01-01 00:00:00');
0afb0c2e (Michael                    2017-02-27 23:37:15 +0000 2074)    $workers = q("SELECT COUNT(*) AS `workers` FROM `workerqueue` WHERE `executed` > '%s'", dbesc(NULL_DATE));
$
````
`NULL_DATE`, maybe use some better OOPed approach like with `DateTime` object? Documentation: http://php.net/datetime

Hypolite commented:
I can't even load the diff on mobile anymore 😅

I'm now completely lost between all the git-fu that @Quix0r did on those PR and the merge conflicts.

About `NULL_DATE`, this is a crutch to avoid to rewrite SQL queries, it has nothing to do with `DateTime`.

Quix0r commented:
@Hypolite Oh, sorry to hear that. :-( Maybe some manual diffing is still possible? e.g. `git diff develop..some-other-branch`.

And the comparison happens in PHP code, too. I will add `@TODO` tags to the merged conflicts.

Quix0r commented:
Done merging.

Quix0r commented:
f**k, parser error ...

Quix0r commented:
Well, common is to have 0000-00-00 00:00:00 as default time stamp when not automatically created, like with statements like `DEFAULT CURRENT_TIMESTAMP`. Documentation: https://dev.mysql.com/doc/refman/5.7/en/timestamp-initialization.html (maybe not part of SQL standard).

Hypolite commented:
The choice of the lowest timestamp as  `0001-01-01 00:00:00` allows full compatibility with MariaDB and PostreSQL.

Quix0r commented:
okay, i was not aware of that change.

annando commented:
You did some bad git foo again. Now you pull request contains 181 files again ...

Quix0r commented:
We really need to get this merged, else it always fall back as it gets rebased. :-( Stupid github interface. :-( They seem to purely depend on the commit ids to show you differences.

Hypolite commented:
Ok, let's do this, after all this only concerns `develop`, and enough people use it that we will be able to pick up on any glaring regression.

annando commented:
Hopefully this hasn't reverted something. I guess I will test this when I have enough time to investigate it ;)

annando commented:
Accidentally I merged it on my local machine - and my system doesn't work anymore.

There is at least one bug in line 2435 of include/dfrn.php

Can we safely revert this?

Hypolite commented:
Safely reverted. Sorry for the mess.

annando commented:
I would vote for restoring the situation when there were only 10 files to merge. These files looked good.

@Quix0r In this line a ```{``` is missing: https://github.com/friendica/friendica/pull/3112/files#diff-08a925e9dc328fe7ae4f3f27479a6678R2427

Are you testing it on your local installation or how is your testing workflow?

Hypolite commented:
I liked it when it was only 10 files to review, and I assumed that the whole 180 files situation only was a byproduct of @Quix0r's git-fu, but apparently it wasn't that clear. I'm still confused.

annando commented:
I'm confused as well. I guess I never will understand this git magic - and I'm still afraid of "rebase" ;-)

@Quix0r is it possible for you to restore the previous state with the 10 files? I guess we can rely on the webinterface. It told something about some 180 files that were changed - and exactly this number of files was merged on my local system as well. So the webinterface seems to show exactly the changed files.

Quix0r commented:
Again 180+ files? :-( But I see it now merged.
